pilot/pkg/bootstrap/server.go

```diff
@@ -1097,6 +1107,14 @@ func (s *Server) maybeCreateCA(caOpts *caOptions) error {
 			if s.RA, err = s.createIstioRA(s.kubeClient, caOpts); err != nil {
 				return fmt.Errorf("failed to create RA: %v", err)
 			}
+			s.environment.AddMeshHandler(func() {
```

Comment:
IMO, this can be done inside the createIstioRA function. This would also require the "SetCACertificatesFromMeshConfig" method to implemented by the RA code and not by the the ca interface

Reply:
In the Mesh handler, I think it might be safer to copy the certificates onto a newer array and use that to "SetCACertificatesFromMeshConfig".

Reply:
This is done inside RA only

Reply:
done